go/obscuronode/enclave/enclave.go

```diff
@@ -575,6 +575,46 @@ func (e *enclaveImpl) GetTransaction(txHash common.Hash) *nodecommon.L2Tx {
 	}
 }
 
+func (e *enclaveImpl) GetTransactionReceipt(encryptedParams nodecommon.EncryptedParams) (nodecommon.EncryptedResponse, error) {
+	paramBytes, err := e.rpcEncryptionManager.DecryptWithEnclaveKey(encryptedParams)
+	if err != nil {
+		return nil, fmt.Errorf("could not decrypt params in eth_getTransactionReceipt request. Cause: %w", err)
+	}
+
+	var paramsJSONMap []string
+	err = json.Unmarshal(paramBytes, &paramsJSONMap)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse JSON params in eth_getTransactionReceipt request. Cause: %w", err)
+	}
+	txHash := common.HexToHash(paramsJSONMap[0]) // The only argument is the transaction hash.
+
+	txReceipt, err := e.storage.GetTransactionReceipt(txHash)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve transaction receipt in eth_getTransactionReceipt request. Cause: %w", err)
+	}
+	txReceiptBytes, err := txReceipt.MarshalJSON()
+	if err != nil {
+		return nil, fmt.Errorf("could not marshall transaction receipt to JSON in eth_getTransactionReceipt request. Cause: %w", err)
+	}
+
+	tx, _, _, _, err := e.storage.GetTransaction(txHash) //nolint:dogsled
```

Comment:
I'd split this up into a few methods 
Quite hard to figure out what's going on here

Reply:
Thanks, done.